src/sparse/KokkosSparse_sptrsv.hpp

```diff
@@ -210,10 +219,584 @@ namespace Experimental {
+  template <typename scalar_type,
+            typename ordinal_type,
+            typename size_type,
+            typename KernelHandle,
+            typename execution_space      = Kokkos::DefaultExecutionSpace,
+            typename host_execution_space = Kokkos::DefaultHostExecutionSpace>
+  void sptrsv_symbolic(
+      KernelHandle *kernelHandleL,
+      KernelHandle *kernelHandleU,
+      SuperMatrix &L,
+      SuperMatrix &U)
+  {
+    typedef KokkosSparse::CrsMatrix<scalar_type, ordinal_type, host_execution_space, void, size_type> host_crsmat_t;
+    typedef KokkosSparse::CrsMatrix<scalar_type, ordinal_type,      execution_space, void, size_type> crsmat_t;
+
+    typedef typename host_crsmat_t::StaticCrsGraphType host_graph_t;
+    typedef typename      crsmat_t::StaticCrsGraphType      graph_t;
+
+    Kokkos::Timer timer;
+    Kokkos::Timer tic;
+
+    // ===================================================================
+    // load sptrsv-handles
+    auto *handleL = kernelHandleL->get_sptrsv_handle ();
+    auto *handleU = kernelHandleU->get_sptrsv_handle ();
+    int *etree = handleL->get_etree ();
+
+    // ===================================================================
+    // load options
+    bool col_majorL = handleL->is_column_major ();
+    bool col_majorU = handleU->is_column_major ();
+    bool merge = handleL->get_merge_supernodes ();
+    bool UinCSC = handleU->is_column_major ();
+    bool needEtree = (handleL->get_algorithm () == SPTRSVAlgorithm::SUPERNODAL_SPMV ||
+                      handleL->get_algorithm () == SPTRSVAlgorithm::SUPERNODAL_ETREE);
+    if (needEtree && etree == nullptr) {
+      std::cout << std::endl
+                << " ** etree needs to be set before calling sptrsv_symbolic with SuperLU **"
+                << std::endl << std::endl;
+      return;
+    }
+
+    // ===================================================================
+    // read CrsGraph from SuperLU factor
+    #ifdef KOKKOS_SPTRSV_SUPERNODE_PROFILE
+    std::cout << " > Read SuperLU factor into KokkosSparse::CrsMatrix (invert diagonal and copy to device)" << std::endl;
+    if (merge) {
+      std::cout << " > Merge supernodes" << std::endl;
+    }
+    #endif
+    bool cusparse = false; // pad diagonal blocks with zeros
+    host_graph_t graphL_host;
+    host_graph_t graphU_host;
+    graph_t graphL;
+    graph_t graphU;
+
+    tic.reset();
+    graphL_host = read_superlu_graphL<host_graph_t> (cusparse, merge, &L);
+    if (UinCSC) {
+      graphU_host = read_superlu_graphU_CSC<host_graph_t> (&L, &U); 
+    } else {
+      graphU_host = read_superlu_graphU<host_graph_t> (&L, &U); 
+    }
+    #ifdef KOKKOS_SPTRSV_SUPERNODE_PROFILE
+    std::cout << "   Conversion Time (from SuperLU to CSR): " << tic.seconds() << std::endl;
+    #endif
+
+    // ===================================================================
+    // setup supnodal info
+    SCformat *Lstore = (SCformat*)(L.Store);
```

Comment:
This looks like a cast from `void*`.  If so, then prefer C++ `reinterpret_cast<SCformat*>(L.Store)` to the C-style cast you did here.